agent/custom/support_action.py

```
@@ -0,0 +1,687 @@
+        roi = _roi(img, bx, by, anchor[0] + DIGIT_OFF[0], anchor[1] + DIGIT_OFF[1],
+                   DIGIT_SIZE[0], DIGIT_SIZE[1])
+        if roi is None:
+            return False
+        got = SupportAction._ocr_skill_text(roi)
+        ok = got is not None and got == str(expect)
+        mfaalog.info(f"[SupportAction] 技能({anchor}) 期望={expect} "
+                     f"OCR识别={got or '(无数字)'} -> {'OK' if ok else 'NO'}")
+        return ok
+
+    # ---------- 宝具等级匹配 ----------
+    def _match_np(self, img, np_dir, bx, by, expect):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_match_np`

(ANN202)